packages/library/src/commonMain/kotlin/io/realm/Realm.kt

```diff
@@ -57,54 +64,86 @@ class Realm private constructor(configuration: RealmConfiguration, dbPointer: Na
         this(configuration, RealmInterop.realm_open(configuration.nativeConfig))
 
     /**
-     * TODO Add docs when this method is implemented.
+     * Modify the underlying Realm file in a suspendable transaction on the default Realm Write
+     * Dispatcher.
+     *
+     * The write transaction always represent the latest version of data in the Realm file, even if
+     * the calling Realm not yet represent this.
+     *
+     * Write transactions automatically commit any changes made when the closure returns unless
+     * [MutableRealm.cancelWrite] was called.
+     *
+     * @param block function that should be run within the context of a write transaction.
+     * @return any value returned from the provided write block. If this is a RealmObject it is
+     * frozen before returning it.
+     * @see [RealmConfiguration.writeDispatcher]
      */
     suspend fun <R> write(block: MutableRealm.() -> R): R {
```

Comment:
Maybe for later, but should the write have the dispatcher as a second parameter or we enforce that people can only change the dispatcher through the configuration?

Reply:
We cannot change the dispatcher per invocation as the `SuspendableWriter` holds it realm instance open on the dispatcher thread.